From 64db81bb87cb7ced503fbb0234f926b1adaaa7d5 Mon Sep 17 00:00:00 2001
From: Laurentiu Palcu <laurentiu.palcu@nxp.com>
Date: Tue, 21 Nov 2017 09:32:25 +0200
Subject: [PATCH 2855/5242] MLK-16922-2: drm: imx: dcss: call DCSS runtime PM
 routines

commit  c78435b7ab4c592fdb90365e0ad3a9f58e466c70 from
https://source.codeaurora.org/external/imx/linux-imx.git

When CRTC is disabled/enabled, the runtime PM routines should be called.

Signed-off-by: Laurentiu Palcu <laurentiu.palcu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/drm/imx/dcss/dcss-crtc.c |   10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/drivers/gpu/drm/imx/dcss/dcss-crtc.c b/drivers/gpu/drm/imx/dcss/dcss-crtc.c
index aeafc46..763e285 100644
--- a/drivers/gpu/drm/imx/dcss/dcss-crtc.c
+++ b/drivers/gpu/drm/imx/dcss/dcss-crtc.c
@@ -16,6 +16,7 @@
 #include <linux/device.h>
 #include <linux/platform_device.h>
 #include <linux/component.h>
+#include <linux/pm_runtime.h>
 #include <drm/drmP.h>
 #include <drm/drm_crtc_helper.h>
 #include <drm/drm_atomic_helper.h>
@@ -169,9 +170,13 @@ static void dcss_crtc_atomic_enable(struct drm_crtc *crtc,
 						   base);
 	struct dcss_soc *dcss = dev_get_drvdata(dcss_crtc->dev->parent);
 
+	pm_runtime_get_sync(dcss_crtc->dev->parent);
+
 	dcss_ss_enable(dcss, true);
 	dcss_dtg_enable(dcss, true);
 	dcss_ctxld_enable(dcss);
+
+	crtc->enabled = true;
 }
 
 static void dcss_crtc_atomic_disable(struct drm_crtc *crtc,
@@ -195,6 +200,11 @@ static void dcss_crtc_atomic_disable(struct drm_crtc *crtc,
 	dcss_ss_enable(dcss, false);
 	dcss_dtg_enable(dcss, false);
 	dcss_ctxld_enable(dcss);
+
+	crtc->enabled = false;
+
+	pm_runtime_mark_last_busy(dcss_crtc->dev->parent);
+	pm_runtime_put_autosuspend(dcss_crtc->dev->parent);
 }
 
 static const struct drm_crtc_helper_funcs dcss_helper_funcs = {
-- 
1.7.9.5

